### PR TITLE
Adding dotnet isolated image to the core-tools nightly build

### LINKED ISSUE #394: Core-Tools Nightly Build fails with DotNet image 5.
https://azure-functions.visualstudio.com/azure-functions-docker/_build/results?buildId=7990&view=logs&j=6c314701-959d-5fd0-416e-8174fd0cc497&t=4d425a9b-3a28-58a6-acb9-2d8644e0627c

```
failed to solve with frontend dockerfile.v0: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount730593061/dotnet-core-tools.Dockerfile: no such file or directory
```

Need investigation, however, it looks .NET 5 related updated is not done on this pipeline.

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
Fix : https://github.com/Azure/azure-functions-docker/issues/394

Since the dotnet-isolated imaged is added, the directory structure has not been changed. 
This Pipeline also add the feature to publish the dotnet 5 isolated core tools images. 

@ankitkumarr Could you have a look?

### CODE REVIEW

File: README.md
Comment (ankitkumarr) [inline README.md]: Thanks for adding this!! :) Ahmed had asked me, but it slipped my mind.

File: host/3.0/core-tools-publish.yml
Comment (ankitkumarr) [inline host/3.0/core-tools-publish.yml]: would this change to `host/3.0/buster/amd64/dotnet/dotnet-isolated/`?

File: host/3.0/core-tools-publish.yml
Comment (ankitkumarr) [inline host/3.0/core-tools-publish.yml]: would this change to `host/3.0/buster/amd64/dotnet/dotnet-inproc/`?

File: host/3.0/core-tools-publish.yml
Comment (ankitkumarr) [inline host/3.0/core-tools-publish.yml]: I think these would be "$SOURCE_REGISTRY/**dotnet-isolated**:$(PrivateVersion)-dotnet-isolated5.0-core-tools". Same throughout this file.

File: README.md
Comment (TsuyoshiUshio) [reply README.md]: You are welcome!

File: host/3.0/core-tools-publish.yml
Comment (TsuyoshiUshio) [reply host/3.0/core-tools-publish.yml]: Done. Very good catch

File: host/3.0/core-tools-publish.yml
Comment (TsuyoshiUshio) [reply host/3.0/core-tools-publish.yml]: Very good catch, I fixed wrong place.  Could you review this place again? @ankitkumarr

File: host/3.0/core-tools-publish.yml
Comment (TsuyoshiUshio) [reply host/3.0/core-tools-publish.yml]: Very good catch. Done

**Review by pragnagopa — APPROVED** [review]
LGTM. Please wait for approval from @ankitkumarr

File: host/3.0/core-tools-publish.yml
Comment (ankitkumarr) [inline host/3.0/core-tools-publish.yml]: Should this and above be "docker tag $SOURCE_REGISTRY/dotnet-isolated:$(PrivateVersion)-dotnet-isolated5.0-core-tools $TARGET_REGISTRY/**dotnet-isolated**:$(MajorVersion)-dotnet-isolated5.0-core-tools"?

**Review by ankitkumarr — APPROVED** [review]
Just one more comment. Looks good otherwise.

File: host/3.0/core-tools-publish.yml
Comment (TsuyoshiUshio) [reply host/3.0/core-tools-publish.yml]: Thanks! good catch!

**Pull request merged**
